tensorflow/core/grappler/optimizers/auto_mixed_precision.cc

```diff
@@ -41,27 +41,35 @@ limitations under the License.
 #include "tensorflow/core/platform/logging.h"
 #include "tensorflow/core/util/env_var.h"
 
+#if TENSORFLOW_USE_ROCM
+#include "rocm/include/hip/hip_runtime.h"
+#endif
+
 namespace tensorflow {
 namespace grappler {
 
 #if TENSORFLOW_USE_ROCM
-const std::array<std::string, 2> FP16SupportedDevices = {"906", "908"}; 
-
-bool HasEnhancedFP16ComputeSupport(std::pair<int, int> gpu_arch){
-    std::string arch = std::to_string(gpu_arch.first); 
-    return std::find(std::begin(FP16SupportedDevices), 
-                     std::end(FP16SupportedDevices), arch)
-           != std::end(FP16SupportedDevices); 
+bool GetFastFP16Support(const DeviceProperties &props)
+{
+    bool supported = false;
+    std::array<std::string, 2> FP16SupportedDevices = {"gfx906", "gfx908"};
```

Comment:
will a std::set<std:string> work better?

Reply:
I'm not sure it matters with the list this small. With std::array it's just a wrapper for a plain-old-data. Not sure if we want the overhead that comes with set for this size of list.

Reply:
> I'm not sure it matters with the list this small

correct...the list is so small, that performance considerations are a no-op. Hence the argument for using std::set for better readability.